### DISCUSSION TITLE
Slow Start

### DISCUSSION THREAD

**Discussion by wsyqn6** [JanDeDobbeleer/oh-my-posh#5066] [Troubleshoot] [answered]
###  Load slow when Terminal open or new tab create every time

> PowerShell 7.4.2
> Loading personal and system profiles took 1332ms.

oh-my-posh ver 
> 21.6.0

os  
> win 11 22H2

`measure-Command { oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/craver.omp.json" }`
> 37ms

`measure-Command { oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/craver.omp.json" | Invoke-Expression}`
>  1180ms

oh-my-posh debug

Version: 21.6.0

Shell: pwsh (7.4.2)

Prompt:

   11:47:41  ~ 
➜

Segments:

ConsoleTitle(true)  -   1 ms
Root(false)         -   0 ms
Os(true)            -   0 ms
Time(true)          -   0 ms
Path(true)          -   1 ms
Git(false)          -   4 ms
Dotnet(false)       -   0 ms
Executiontime(false) -   0 ms
Status(false)       -   1 ms
Text(true)          -   0 ms

Run duration: 17.685ms

Cache path: C:\Users\19081\AppData\Local\oh-my-posh

Config path: C:\Users\19081\AppData\Local\Programs\oh-my-posh\themes\craver.omp.json

thanks for your help 😊

**Comment by JanDeDobbeleer** [comment]
@a641390597 that's PowerShell being slow to run Invoke-Expression. You can check how long the command actually takes by running 

```powershell
Measure-Command { oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/craver.omp.json" --print }
```

<img width="866" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/2492783/95869b39-24f8-4e18-8859-dde68076ed90">

With `Invoke-Expression`:
<img width="931" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/2492783/90376e24-4785-4987-bf93-4bb6f421d1bd">

**Reply by wsyqn6** [reply]
`Measure-Command { oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/craver.omp.json" --print | Invoke-Expression}`
>TotalMilliseconds : 117.494

**Reply by JanDeDobbeleer** [reply]
See, that's not > 1 second. So it's not oh-my-posh being slow here.

**Reply by wsyqn6** [reply]
`Measure-Command { oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/craver.omp.json" | Invoke-Expression}`
> TotalMilliseconds : 1176.1918

last reply is wirte wrong because  network failure

**Reply by JanDeDobbeleer** [reply, marked as answer]
@a641390597 Have a look at what could block oh-my-posh execution like Defender or any other anti virus tool.
